src/macaron/policy_engine/prelude/aggregate_rules.dl

```diff
@@ -0,0 +1,87 @@
+/* Copyright (c) 2023 - 2023, Oracle and/or its affiliates. All rights reserved. */
+/* Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/. */
+
+/**
+ *  Souffle datalog rules defining an aggregate approximate security level for a repository.
+ *  This is not a SLSA level, but is based on Macaron checks which are inspired by SLSA requirements.
+ *
+ *  For a repository to meet one level, it must meet the requirements of all lower levels.
+ *  These levels do consider the level of a repository's dependencies.
+ */
+
+/* aggregate verifying a level */
+.decl aggregate_level(level: number, repo: number)
```

Comment:
How are `aggregate_level` and `aggregate_level_requirement` differ from each other? :thinking:

Reply:
`aggregate_level` considers the dependencies of `repo` as well

Reply:
Would it be okay to clarify it using a short comment?

Reply:
Clarified in https://github.com/oracle-samples/macaron/pull/46/commits/ee565267d152c2e1439bec8ed96c0730d4fc8f6d